migrations/2025-02-05-083109_add-num-versions-column/up.sql

```diff
@@ -0,0 +1,26 @@
+ALTER TABLE default_versions
+    ADD COLUMN num_versions INTEGER NOT NULL DEFAULT 0;
+
+COMMENT ON COLUMN default_versions.num_versions IS 'The total number of versions.';
+
+CREATE OR REPLACE FUNCTION update_num_versions_from_versions() RETURNS TRIGGER AS $$
+BEGIN
+    IF (TG_OP = 'INSERT') THEN
+        UPDATE default_versions
+        SET num_versions = num_versions + 1
+        WHERE crate_id = NEW.crate_id;
```

Comment:
I'm wondering if we should use `INSERT INTO default_versions ON CONFLICT (crate_id) DO UPDATE SET` here. that should allow us to completely remove the `diesel::insert_into(default_versions::table)` part from the publish endpoint and also set the `num_versions: 1` default here at the same time.

Reply:
I also considered this previously :D I am fine with this.

Reply:
updated!